### PR TITLE
STM32F103xx & STM32F40xx  SD card simulates EEPROM to store data

### PR DESCRIPTION
**Pull request by yangwenxiong**
### Requirements

1.Thanks for the previous request. Someone helped with the optimization, but there were new problems after the optimization, so I made the final arrangement 
2.
### Description

1.In Marlin-bugfix-2.0.x -A\Marlin\src\HAL\HAL_STM32\persistent_store_sdcard.cpp modification

      #define HAL_STM32F4_EEPROM_SIZE (E2END + 1) // 16KB
     #define _ALIGN(x) __attribute__ ((aligned(x))) // SDIO uint32_t* compat.
     static char _ALIGN(4) HAL_eeprom_data[HAL_STM32F4_EEPROM_SIZE];
2. In Marlin-bugfix-2.0.x -A\Marlin\src\module\configuration_store.cpp Restores some modifications

### Benefits

1.For the MCU of STM32F103xx series, it can solve the need to create a new EEPROM.DAT file in SD card in advance to use the phenomenon of EEPROM simulation
2.Welcome engineers to verify, thank you

### CODE REVIEW

File: Marlin/src/HAL/HAL_STM32/persistent_store_sdcard.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+/**
+ * HAL for stm32duino.com based on Libmaple and compatible (STM32F1)
+ * Implementation of EEPROM settings in SD Card
+ */
+
+#ifdef TARGET_STM32F4
+
+#include "../../inc/MarlinConfig.h"
+
+#if ENABLED(EEPROM_SETTINGS) && NONE(FLASH_EEPROM_EMULATION, SPI_EEPROM, I2C_EEPROM)
```
Comment (thinkyhead) [inline Marlin/src/HAL/HAL_STM32/persistent_store_sdcard.cpp]: What about `SRAM_EEPROM_EMULATION` ?

Comment (yangwenxiong) [reply Marlin/src/HAL/HAL_STM32/persistent_store_sdcard.cpp]: Good chaos

Comment (yangwenxiong) [reply Marlin/src/HAL/HAL_STM32/persistent_store_sdcard.cpp]: This can become what you modified before

Comment (yangwenxiong) [reply Marlin/src/HAL/HAL_STM32/persistent_store_sdcard.cpp]: @thinkyhead This file simulates eeprom for STM32F40xx external SD card

**Comment by thinkyhead** [comment]
After doing a merge with the current `bugfix-2.0.x` I don't see that this adds anything new.

File: Marlin/src/HAL/HAL_STM32/persistent_store_sdcard.cpp (hunk 1 above)
Comment (yangwenxiong) [reply Marlin/src/HAL/HAL_STM32/persistent_store_sdcard.cpp]: 使用SRAM_EEPROM_EMULATION ，这个是在Marlin-bugfix-2.0.x -A\Marlin\src\HAL\HAL_STM32\persistent_store_impl.cpp
#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC)

#include "../../inc/MarlinConfig.h"

#if ENABLED(EEPROM_SETTINGS) && ANY(FLASH_EEPROM_EMULATION, SRAM_EEPROM_EMULATION, SPI_EEPROM, I2C_EEPROM)

**Comment by tpruvot** [comment]
fix is in PR #14844

File: Marlin/src/HAL/HAL_STM32/persistent_store_sdcard.cpp (hunk 1 above)
Comment (yangwenxiong) [reply Marlin/src/HAL/HAL_STM32/persistent_store_sdcard.cpp]: Use SRAM_EEPROM_EMULATION, which was in marlin-bugfix 2.0. X-a \Marlin\ SRC \HAL\HAL_STM32\persistent_store_impl.cpp

#if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC)

#include "../../inc/MarlinConfig.h"

#if ENABLED(EEPROM_SETTINGS) && ANY(FLASH_EEPROM_EMULATION, SRAM_EEPROM_EMULATION, SPI_EEPROM, I2C_EEPROM)

**Pull request closed without merging**